src/main/java/pl/mantiscrab/budgetr/registration/UserController.java

```diff
@@ -0,0 +1,35 @@
+package pl.mantiscrab.budgetr.registration;
+
+import lombok.AllArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import pl.mantiscrab.budgetr.registration.dto.UserDto;
+import pl.mantiscrab.budgetr.registration.dto.UserRegisterDto;
+import pl.mantiscrab.budgetr.registration.exception.ConstraintViolation;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@AllArgsConstructor
+@RestController
+class UserController {
+    private final UserService service;
+
+    @PostMapping("/register")
+    ResponseEntity<UserDto> register(@Valid @RequestBody UserRegisterDto userRegisterDto) {
+        UserDto userDto = service.register(userRegisterDto);
+        return ResponseEntity.ok().body(userDto);
+    }
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    List<ConstraintViolation> handleMethodArgumentNotValidException(MethodArgumentNotValidException ex) {
```

Comment:
Good that you remembered about it :-) 

If you wish to see how I implemented that generic exception you can follow the page:
https://github.com/greencashew/spring-link-shortener/tree/main/src/main/java/dev/greencashew/link_shortener/configuration/web/exception_handling